server/src/main/java/org/candlepin/model/OwnerProductCurator.java

```diff
@@ -595,15 +594,19 @@ public void updateOwnerProductReferences(Owner owner, Map<String, String> produc
             .list();
 
         if (ids != null && !ids.isEmpty()) {
-            criteria.clear();
-            criteria.put("product_uuid", productUuidMap.keySet());
-            criteria.put("pool_id", ids);
-
-            count = this.bulkSQLUpdate("cp2_pool_provided_products", "product_uuid", uuidMap, criteria);
-            log.debug("{} provided products updated", count);
-
-            count = this.bulkSQLUpdate("cp2_pool_derprov_products", "product_uuid", uuidMap, criteria);
-            log.debug("{} derived provided products updated", count);
+            for (List<String> poolIdBlock : this.partition(ids, getBatchBlockSize())) {
+                Map<String, Object> updateCriteria = new HashMap<>();
+                updateCriteria.put("product_uuid", productUuidMap.keySet());
+                updateCriteria.put("pool_id", poolIdBlock);
+
+                int providedProductsUpdated = this.bulkSQLUpdate("cp2_pool_provided_products",
+                    "product_uuid", uuidMap, updateCriteria);
+                log.debug("{} provided products updated", providedProductsUpdated);
+
+                int derivedProductsUpdated = this.bulkSQLUpdate("cp2_pool_derprov_products", "product_uuid",
+                    uuidMap, updateCriteria);
+                log.debug("{} derived provided products updated", derivedProductsUpdated);
```

Comment:
Sorry, I guess I wasn't clear on this last time. I meant moving the output statements to outside the loop rather than a bunch of individual updates that will elongate output and require manual math during debugging.

Reply:
Done.